plugins/modules/idrac_reset.py

```diff
@@ -92,41 +169,312 @@
+                                                                              max_job_wait_sec=self.job_wait_timeout,
+                                                                              sleep_interval_secs=1)
+            job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+            if job_failed:
+                self.module.exit_json(msg=job_dict.get("Message"), job_status=job_dict, failed=True)
+
+    def wait_for_port_open(self, interval=30):
+        timeout_wait = self.module.params.get('job_wait_timeout')
+        time.sleep(interval // 2)
+        msg = RESET_UNTRACK
+        wait = timeout_wait
+        track_failed = True
+        status_code = 503
+        while int(wait) > 0:
+            try:
+                self.idrac.invoke_request(MANAGERS_URI, 'GET')
+                time.sleep(interval // 2)
+                msg = IDRAC_RESET_SUCCESS_MSG
+                track_failed = False
+                status_code = 200
+                break
+            except HTTPError as err:
+                status_code = err.code
+                if status_code == 401:
+                    time.sleep(interval // 2)
+                    msg = IDRAC_RESET_SUCCESS_MSG
+                    track_failed = False
+                    break
+            except URLError as err:
+                time.sleep(interval)
+                wait = wait - interval
+        return track_failed, status_code, msg
+
+    def reset_to_default_mapped(self):
+        payload = {}
+        payload["ResetType"] = self.reset_to_default
+        self.validate_obj.validate_reset_options(self.allowed_choices, RESET_KEY)
+        return self.perform_operation(payload)
+
+    def reset_custom_defaults(self):
+        # add a check for firmware version for this > 7.00.30.00
+        if self.idrac_firmware_version < '7.00.30.00':
+            self.module.exit_json(msg=RESET_TO_DEFAULT_ERROR.format(self.reset_to_default, self.allowed_choices), skipped=True)
+        tmp_res, res = None, None
+        custom_default_file = self.module.params.get('custom_defaults_file')
+        custom_default_buffer = self.module.params.get('custom_defaults_buffer')
+        upload_perfom = False
+        default_data = None
+        if custom_default_file:
+            self.validate_obj.validate_path(custom_default_file)
+            self.validate_obj.validate_file_format(custom_default_file)
+            upload_perfom = True
+            default_data = custom_default_file
+        elif custom_default_buffer:
+            upload_perfom = True
+            default_data = custom_default_buffer
+        if upload_perfom:
+            self.upload_cd_file(default_data)
+        self.validate_obj.validate_custom_option(self.reset_to_default, self.allowed_choices)
+        return self.reset_to_default_mapped()
+
+    def graceful_restart(self):
+        payload = {}
+        payload["ResetType"] = "GracefulRestart"
```

Comment:
Define this in one line: 
 payload = {"ResetType": "GracefulRestart"}

Reply:
Done